### PR TITLE
Finish "Enable checking keys for equality"

### LINKED ISSUE #134: Add __eq__() magic method to key classes
I'm using public keys as identifiers and need a way to check two such identifiers for equality. Now, the way the keys are stored internally by PyNaCl varies among the different key classes (`nacl.public.PublicKey`, `nacl.public.VerifyKey`, `nacl.public.SigningKey`, …) and I really don't want to rely on that. Hence, a magic method `__eq__()` would be a great thing to have.

I will probably add it myself in the coming days and create a pull request. Just wanted to write it down in the meantime.

### PR DESCRIPTION
**Pull request by alexwlchan**
This picks up where #136 left off, in particular adding `__ne__` methods and a bit more testing.

With apologies for stealing @codethief‘s thunder, but after eighteen months I thought it would probably be okay to pick it up and finish it off.

This fixes #134.

### CODE REVIEW

**Comment by alexwlchan** [comment]
Sheesh, coveralls is keen.

**Comment by lmctv** [comment]
I'm thinking  that it would be better if this used the bindings for sodium_memcmp. If you would like so, I'll try to send a pull request to this branch (after the review cycle for blake2b hash PRs).

File: src/nacl/public.py [hunk 1]
```diff
@@ -81,6 +89,14 @@ def __init__(self, private_key, encoder=encoding.RawEncoder):
     def __bytes__(self):
         return self._private_key
 
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return bytes(self) == bytes(other)
```
Comment (reaperhulk) [inline src/nacl/public.py]: As @lmctv suggested @warner and I would like to see this be a constant time comparison using `sodium_memcp`. Other than that this is fine.

Comment (reaperhulk) [inline src/nacl/public.py]: Thanks for bearing with us on this btw, I apologize for the long, long (long) wait.

Comment (lmctv) [inline src/nacl/public.py]: I'm preparing a PR for @alexwlchan  based on this branch. Hope it can be of use.

Comment (reaperhulk) [inline src/nacl/public.py]: @lmctv if you want to submit a PR that cherry-picks this + adds the fixes requested we can review that ASAP

**Comment by alexwlchan** [comment]
Superseded by #232.

**Pull request closed without merging**